src/main/java/com/google/vinet/servlets/LoginServlet.java

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.vinet.servlets;
+
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.users.UserService;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet that returns a login link, which when visited redirects the user to
+ * the provided redirection URL.
+ */
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+  private UserService userService;
+
+  /**
+   * Construct a LoginServlet that depends on the default implementaion of UserService.
+   */
+  public LoginServlet() {
+    this.userService = UserServiceFactory.getUserService();
+  }
+
+  /**
+   * Construct a LoginServlet that depends on the provided implementation of UserService..
+   * @param userService The implementation of UserService the LoginServlet will depend on.
+   */
+  public LoginServlet(UserService userService) {
+    this.userService = userService;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    if (response == null) {
+      throw new IllegalArgumentException("response cannot be null");
+    }
+
+    if (request == null) {
+      throw new IllegalArgumentException("request cannot be null");
+    }
```

Comment:
Do these return an error code to the frontend at all?

Reply:
These resources aren't created by the frontend, but instead by App Engine, so it makes most sense to report these errors to the caller.

The thought process was:
* In the case of the `response` being null, responding to the frontend is impossible, so passing the error to the caller is the only meaningful option.
* In the case of the `request` being null, since this resource is created by App Engine when it calls `doGet()`, the error should be passed up to the caller.

Reply:
I believe `response` is internal annotated with @NotNull, so it should never really be null

Reply:
I'm not sure I want to take the risk, and end up in the `// This should never happen` hall of fame. No harm in keeping a check that ultimately should never never fail.